hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -1883,45 +1883,59 @@ func taintsToJSON(taints []hyperv1.Taint) (string, error) {
 	return string(taintsInJSON), nil
 }
 
-func (r *NodePoolReconciler) reconcileMachineHealthCheck(ctx context.Context,
-	mhc *capiv1.MachineHealthCheck,
-	nodePool *hyperv1.NodePool,
-	hc *hyperv1.HostedCluster,
-	CAPIClusterName string) error {
+func (r *NodePoolReconciler) reconcileMachineHealthCheck(ctx context.Context, mhc *capiv1.MachineHealthCheck, nodePool *hyperv1.NodePool,
+	hc *hyperv1.HostedCluster, CAPIClusterName string) error {
 
 	log := ctrl.LoggerFrom(ctx)
 
-	// Opinionated spec based on
+	// Opinionated defaults based on
 	// https://github.com/openshift/managed-cluster-config/blob/14d4255ec75dc263ffd3d897dfccc725cb2b7072/deploy/osd-machine-api/011-machine-api.srep-worker-healthcheck.MachineHealthCheck.yaml
-	// TODO (alberto): possibly expose this config at the nodePool API.
-	maxUnhealthy := intstr.FromInt(2)
-	timeOut := 8 * time.Minute
-
-	maxUnhealthyOverride := nodePool.Annotations[hyperv1.MachineHealthCheckMaxUnhealthyAnnotation]
-	if maxUnhealthyOverride == "" {
-		maxUnhealthyOverride = hc.Annotations[hyperv1.MachineHealthCheckMaxUnhealthyAnnotation]
+	maxUnhealthy := ptr.To(intstr.FromInt(2))
+	var unhealthyRange *string
+	timeout := &metav1.Duration{
+		Duration: 8 * time.Minute,
+	}
+	nodeStartupTimeout := &metav1.Duration{
+		Duration: 20 * time.Minute,
 	}
+
+	maxUnhealthyOverride := hc.Annotations[hyperv1.MachineHealthCheckMaxUnhealthyAnnotation]
 	if maxUnhealthyOverride != "" {
 		maxUnhealthyValue := intstr.Parse(maxUnhealthyOverride)
 		// validate that this is a valid value by getting a scaled value
 		if _, err := intstr.GetScaledValueFromIntOrPercent(&maxUnhealthyValue, 100, true); err != nil {
 			log.Error(err, "Cannot parse max unhealthy override duration", "value", maxUnhealthyOverride)
 		} else {
-			maxUnhealthy = maxUnhealthyValue
+			maxUnhealthy = &maxUnhealthyValue
 		}
 	}
 
-	timeOutOverride := nodePool.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation]
-	if timeOutOverride == "" {
-		timeOutOverride = hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation]
-	}
-	if timeOutOverride != "" {
-		timeOutOverrideTime, err := time.ParseDuration(timeOutOverride)
+	timeoutOverride := hc.Annotations[hyperv1.MachineHealthCheckTimeoutAnnotation]
```

Comment:
do we need to keep those override annotations? @csrwng @enxebre

Reply:
yes, we need to keep it for now to enable a smooth transition in future.